examples/vm-from-user-image-managed-disk/main.tf

```diff
@@ -0,0 +1,116 @@
+# Configure the Microsoft Azure Provider
+# ARM_SUBSCRIPTION_ID, ARM_CLIENT_ID, ARM_CLIENT_SECRET and ARM_TENANT_ID environment variables assumed to be set 
+# If environment variables not set, declare in azurerm provider
+provider "azurerm" {}
+
+variable "publicIPAddressType" {
+  default = "Dynamic"
+}
+
+variable "osTypeMap" {
+  description = "This is the OS that your VM will be running (value constraints)"
+  type        = "map"
+
+  default = {
+    Windows = "windows"
+    Linux   = "linux"
+  }
+}
+
+variable "newOrExistingVnetMap" {
+  description = "Select if this template needs a new VNet or will reference an existing VNet (value constraints)"
+  type        = "map"
+
+  default = {
+    new      = "1"
+    existing = "0"
+  }
+}
+
+resource "azurerm_resource_group" "resource_group" {
+  name     = "${var.resourceGroupName}"
+  location = "${var.resourceGroupLocation}"
+}
+
+resource "azurerm_virtual_network" "virtual_network1" {
+  # Create only if New VNet is specified 
+  count               = "${lookup(var.newOrExistingVnetMap, var.newOrExistingVnet)}"
+  name                = "${var.newOrExistingVnetName}"
+  address_space       = ["10.0.0.0/16"]
+  location            = "${var.resourceGroupLocation}"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+}
+
+resource "azurerm_subnet" "subnet1" {
+  # Create only if New VNet is specified 
+  count                = "${lookup(var.newOrExistingVnetMap, var.newOrExistingVnet)}"
+  name                 = "${var.newOrExistingSubnetName}"
+  resource_group_name  = "${azurerm_resource_group.resource_group.name}"
+  virtual_network_name = "${azurerm_virtual_network.virtual_network1.name}"
+  address_prefix       = "10.0.0.0/24"
+}
+
+resource "azurerm_public_ip" "public_ip1" {
+  name                         = "${var.customVmName}IP"
+  location                     = "${var.resourceGroupLocation}"
+  resource_group_name          = "${azurerm_resource_group.resource_group.name}"
+  public_ip_address_allocation = "${var.publicIPAddressType}"
+  domain_name_label            = "${var.dnsLabelPrefix}"
+}
+
+resource "azurerm_network_interface" "network_interface1" {
+  # Need Explicit Dependency for subnet as we don't reference the ID as it may already exist
+  depends_on          = ["azurerm_subnet.subnet1"]
+  name                = "${var.customVmName}Nic"
+  location            = "${var.resourceGroupLocation}"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+
+  ip_configuration {
+    name                          = "ipconfig1"
+    subnet_id                     = "${azurerm_resource_group.resource_group.id}/providers/Microsoft.Network/virtualNetworks/${var.newOrExistingVnetName}/subnets/${var.newOrExistingSubnetName}"
```

Comment:
as such, could we reference the subnet by it's ID, rather than having the `depends_on` here?